kotlinlib/src/mill/kotlinlib/android/AndroidHiltSupport.scala

```diff
@@ -0,0 +1,331 @@
+package mill.kotlinlib.android
+
+import coursier.Repository
+import mill.api.{PathRef, Result}
+import mill.define.{Discover, ExternalModule, ModuleRef}
+import mill.javalib.android.AndroidSdkModule
+import mill.kotlinlib.DepSyntax
+import mill.kotlinlib.android.AndroidHiltSupport.HiltGeneratedClasses
+import mill.kotlinlib.ksp.KspModule
+import mill.kotlinlib.worker.api.KotlinWorkerTarget
+import mill.scalalib.{Dep, JvmWorkerModule}
+import mill.{T, Task}
+import os.Path
+
+import java.io.File
+
+@mill.api.experimental
+trait AndroidHiltSupport extends KspModule with AndroidAppKotlinModule {
+
+  override def sources: T[Seq[PathRef]] = Task { Seq.empty[PathRef] }
+
+  override def kspSources: T[Seq[PathRef]] = Task { super.sources() }
+
+  override def generatedSources: T[Seq[PathRef]] =
+    super[AndroidAppKotlinModule].generatedSources
```

Comment:
Don't reused tasks, use their results.

```suggestion
    super[AndroidAppKotlinModule].generatedSources()
```